tests/serial/time/implicit/serial-coupling/ReadWriteScalarDataFirstParticipantHighTolerance.cpp

```diff
@@ -88,30 +89,36 @@ BOOST_AUTO_TEST_CASE(ReadWriteScalarDataFirstParticipantHighTolerance)
       nbrTimeWindows++;
     }
 
-    dt = solverDt > preciceDt ? preciceDt : solverDt;
+    preciceDt = precice.getMaxTimeStepSize();
+
+    if (abs(preciceDt - solverDt) < minTimeStepSize) {
+      dt = preciceDt;
+    } else {
+      dt = solverDt > preciceDt ? preciceDt : solverDt;
+    }
```

Comment:
This is the main thing our users would have to add in their adapter codes, if they have a time step restriction. Main argument here why removing the feature makes sense: preCICE cannot really know about a suitable strategy, a user would apply, if `preciceDt` of the next time step becomes smaller than the minimum allowed `solverDt`. Therefore, let the user detect the problematic situation and deal with the problem.

Reply:
I think we could technically delete this test as well, since it does not test a specific preCICE feature anymore and the features it tests are tested better elsewhere. It might however make sense to add this check to the other test cases as whit minTimeStepSize=1e-14 to safeguard against time drift and tripping the error in  #1867.